ci/tools/gen_pow10_table.py

```
@@ -0,0 +1,91 @@
+
+def main() -> int:
+    entries: list[tuple[int, int]] = []
+    for k in range(K_MIN, K_MAX + 1):
+        v = Decimal(10) ** k
+        m, e = normalize(v)
+        entries.append((m, e))
+
+    print("// AUTO-GENERATED by ci/tools/gen_pow10_table.py — do not edit by hand.")
+    print(f"// Table indexed by (k - {K_MIN}); k in [{K_MIN}, {K_MAX}].")
+    print(
+        f"// Each entry encodes 10**k ~= mantissa * 2**binary_exp with the"
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)